testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java

```diff
@@ -522,13 +522,20 @@ private ExecutionResult execute(Session session, String query, Optional<String>
                 String temporaryTablePrefix = temporaryTableNamePrefix(queryId);
                 MaterializedResult temporaryTablesResult = getQueryRunner()
                         .execute("SHOW TABLES LIKE '%s%%' ESCAPE '\\'".formatted(temporaryTablePrefix.replace("_", "\\_")));
-                assertThat(temporaryTablesResult.getRowCount())
-                        .as("There should be no remaining %s* tables. They are: [%s]",
-                                temporaryTablePrefix,
-                                temporaryTablesResult.getMaterializedRows().stream()
-                                        .map(row -> row.getField(0).toString())
-                                        .collect(joining(",")))
-                        .isEqualTo(0);
+                // Unfortunately, information_schema is not strictly consistent with recently dropped tables,
+                // and for some connectors, it can return tables that have been recently dropped. Therefore,
+                // we can't rely simply on SHOW TABLES LIKE returning no results - we have to try to query the table
+                for (MaterializedRow temporaryTableRow : temporaryTablesResult.getMaterializedRows()) {
+                    String temporaryTableName = (String) temporaryTableRow.getField(0);
+                    assertThatThrownBy(() -> getQueryRunner().execute("SELECT 1 FROM %s WHERE 1 = 0".formatted(temporaryTableName)))
+                            .as("There should be no remaining %s* tables, but %s is still queryable. The others are: [%s]",
+                                    temporaryTablePrefix,
+                                    temporaryTableName,
+                                    temporaryTablesResult.getMaterializedRows().stream()
+                                            .map(row -> row.getField(0).toString())
+                                            .collect(joining(",")))
+                            .hasMessageContaining("%s does not exist", temporaryTableName);
```

Comment:
Let's extract this logic into a protected method to be overridden where required with the default impl being what you've provided here. I don't think there is a one-size fits all solution.

Reply:
@losipiuk also called out that this shouldn't be done here because it's within a place that is already making assertions, so we get really weird, nested assertion errors when this fails (e.g. in a nested `assertThatThrownBy`).

I am also working on extracting this out into a better location so it's not nested within other assertions, and this whole process will be defined in a protected method so others can override it.

Reply:
@hashhar please review